services/src/main/java/org/keycloak/services/managers/DefaultBruteForceProtector.java

```diff
@@ -208,18 +181,43 @@ protected void failure(KeycloakSession session, LoginEvent event) {
         logger.debugv("waitSeconds: {0}", waitSeconds);
         logger.debugv("deltaTime: {0}", deltaTime);
 
+        boolean quickLoginFailure = false;
         if (waitSeconds == 0) {
             if (last > 0 && deltaTime < realm.getQuickLoginCheckMilliSeconds()) {
                 logger.debugv("quick login, set min wait seconds");
                 waitSeconds = realm.getMinimumQuickLoginWaitSeconds();
+                quickLoginFailure = true;
             }
         }
         if (waitSeconds > 0) {
-            waitSeconds = Math.min(realm.getMaxFailureWaitSeconds(), waitSeconds);
-            int notBefore = (int) (currentTime / 1000) + waitSeconds;
-            logger.debugv("set notBefore: {0}", notBefore);
-            userLoginFailure.setFailedLoginNotBefore(notBefore);
-            sendEvent(session, realm, userLoginFailure, EventType.USER_DISABLED_BY_TEMPORARY_LOCKOUT);
+            if(!realm.isPermanentLockout() || realm.getMaxTemporaryLockouts() > 0) {
```

Comment:
I am in doubt if using a boolean for this is a good approach. WDYT? In the future, this `!realm.isPermanentLockout()` may be difficult to understand since we no longer have two strategies (permanent and !permanent / temporary) but three (permanent, temporary, and mixed) so what means !permanent? It could be permanent or permanent and mixed.  

So perhaps having a method which returns currently lockout strategy would be great. Something like:

```
private String curLockoutStratety() {
    //some checks
    return "PERMANENT"; // or "TEMPORARY" or "MIXED"
}
```

So we will be able to check this in a more clear way:

```
String strategy = curLockoutStratety();
if("TEMPORARY_LOCKOUT".equals(strategy)) 
    //do something
```

WDYT @douglaspalmer ?

Reply:
@pedroigor and I discussed this yesterday. We decided it was better to get this merged and consider refactoring for better readability later.

Reply:
Great.

Reply:
Yeah, we can improve a lot more the code but for now, we want to focus on building on top of it.